src/js_lexer.zig

```diff
@@ -1921,120 +1913,132 @@ fn NewLexer_(
+                                    // 'current' is already after the #/@ trigger.
+                                    lexer.current += offset;
+                                    // Do NOT consume the character immediately after the pragma.
+                                    // Let the main loop find the actual line terminator.
+
+                                    // Continue the outer loop from the position AFTER the pragma arg.
+                                    continue;
+                                }
+                                // If offset == 0, it wasn't a valid pragma start.
+                            }
+                            // Not a pragma or is_json. Treat #/@ as a normal comment character.
+                            // The character was consumed by step(). Let the outer loop continue.
+                            continue;
+                        },
+                        else => {
+                            // Non-ASCII (but not LS/PS), etc. Treat as normal comment char.
+                            // The character was consumed by step(). Let the outer loop continue.
+                            continue;
+                        },
+                    }
+                } else { // Highway found nothing until EOF
+                    // Consume the rest of the line.
+                    lexer.end = lexer.source.contents.len;
+                    lexer.current = lexer.source.contents.len;
+                    lexer.code_point = -1; // Set EOF state
+                    return;
+                }
+            }
+            unreachable;
+        }
+        /// Scans the string for a pragma.
+        /// offset is used when there's an issue with the JSX pragma later on.
+        /// Returns the byte length to advance by if found, otherwise 0.
+        fn scanPragma(lexer: *LexerType, offset_for_errors: usize, chunk: string, allow_newline: bool) usize {
+            if (!lexer.has_pure_comment_before) {
+                if (strings.hasPrefixWithWordBoundary(chunk, "__PURE__")) {
+                    lexer.has_pure_comment_before = true;
+                    return "__PURE__".len;
+                }
+            }
+
+            if (strings.hasPrefixWithWordBoundary(chunk, "jsx")) {
+                if (PragmaArg.scan(.skip_space_first, lexer.start + offset_for_errors, "jsx", chunk, allow_newline)) |span| {
+                    lexer.jsx_pragma._jsx = span;
+                    return "jsx".len +
+                        if (span.range.len > 0) @as(usize, @intCast(span.range.len)) else 0;
+                }
+            } else if (strings.hasPrefixWithWordBoundary(chunk, "jsxFrag")) {
+                if (PragmaArg.scan(.skip_space_first, lexer.start + offset_for_errors, "jsxFrag", chunk, allow_newline)) |span| {
+                    lexer.jsx_pragma._jsxFrag = span;
+                    return "jsxFrag".len +
+                        if (span.range.len > 0) @as(usize, @intCast(span.range.len)) else 0;
```

Comment:
would need a lot of review, but in the future it looks possible to change `Range.len` to be unsigned